tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_graphics_TextLayout.java

```diff
@@ -547,6 +547,60 @@ public void test_setStyle() {
 	assertNull(layout.getStyle(4));
 	assertNull(layout.getStyle(5));
 	layout.dispose();
+
+	layout = new TextLayout (display);
```

Comment:
The test with these changes succeeds on master, so it's not a regression test for the changes made in this PR. Can you adapt the test so that it serves as a regression test for the addressed issue?

Reply:
Here we have not changed the behavior of the api's. The original behavior of the api's still work the same. We are internally detecting, correcting and passing the corrected content in specific cases only. We dint even add new api's. It is only the look and feel by which you can make out if some extra lines are expected or not expected varies. - is my understanding.

Reply:
Okay, I mistakenly expected the added test to be a regression test for the fix because of https://github.com/eclipse-platform/eclipse.platform.swt/pull/1320#issuecomment-2396659597

Still, this change is no refactoring but affects the behavior of public APIs (e.g., `TextLayout#draw(...)` behaves differently). I guess that, e.g., also `TextLayout#getBounds()` now behaves differently, so wouldn't it be possible to have a regression test comparing the bounds of two configurations that should actually result in the bounds but did not do before this change (or vice versa)?

Reply:
@HeikoKlare 
As per your suggestion i have added new junit tests with `TextLayout#getLineCount()` and `TextLayout#getBounds()`.
Since this change is only with respect to windows, i have excluded mac, gtk for the tests to run.

I guess this makes it all complete as we discussed. If i missed out any, please do let me know.

Can you review it and take it forward please?

Reply:
Please hold on, let me cross check and update you again.

Reply:
Hi @HeikoKlare 
Sorry for the confusion caused.
I am successfull in adding only TextLayout#getLineCount() junit. I think we can go ahead with this pr now.

I have tried even TextLayout#getBounds() but had to withdraw as the behavior is not consistent across.
With the sample input taken, 
- On master, getBounds() is returning {0, 0, 30, 220} where width and height are 30, 220 respectively. [junit is failing]
- On the branch where i committed my changes(fix_for_issue184), getBounds() is returning {0, 0, 43, 140}. [junit is passing]
- To cross check, i have pulled the pr 1320 to local, getBounds() is returning {0, 0, 43, 140}. [junit is passing]
- But on the build machine, getBounds() is returning {0, 0, 34, 105}. [junit is failing because it is getting 34 but expecting 43]. -> https://github.com/eclipse-platform/eclipse.platform.swt/actions/runs/12312890737/job/34365830603?pr=1320
Neither able to understand the reason for the failure(to fix) nor able to replicate the problem locally to debug further.

This completes my analysis on this pr. Can you please review now?